addons/sale_pdf_quote_builder/wizards/sale_pdf_quote_builder_dynamic_fields_wizard.py

```diff
@@ -0,0 +1,134 @@
+                'sale_pdf_quote_builder.form_field_path_mapping', json.dumps(new_mapping)
+            )
+
+
+class SalePDFQuoteBuilderDynamicFieldsWizardLine(models.TransientModel):
+    _name = 'sale.pdf.quote.builder.dynamic.fields.wizard.line'
+    _description = "SalePdfQuoteBuilderDynamicFieldsWizardLine transient representation"
+
+    dynamic_fields_wizard_id = fields.Many2one(
+        'sale.pdf.quote.builder.dynamic.fields.wizard', required=True, ondelete='cascade'
+    )
+    document_type = fields.Selection(
+        string="Document Type",
+        selection=[('header_footer', "Header/Footer"), ('product_document', "Product Document")],
+        default='header_footer',
+        required=True,
+    )
+    form_field = fields.Char(
+        string="Form Field", help="Form Field Name as seen in the PF form", required=True,
+    )
+    path = fields.Char(default='', help="Odoo path to get the expected value to fill the PDF.")
+
+    @api.constrains('form_field')
+    def _check_form_field_name_follows_pattern(self):
+        """ Ensure the paths only contains alphanumerics, hyphens and underscores.
+
+        :return: None
+        :raises: ValidationError if the names aren't alphanumerics, hyphens and underscores.
+        """
+        name_pattern = re.compile(r'^(\w|-)+$')
+        for line in self:
+            if not re.match(name_pattern, line.form_field):
+                raise ValidationError(_(
+                    "Invalid form field name %(field_name)s. It should only contain alphanumerics,"
+                    " hyphens or underscores.",
+                    field_name=line.form_field,
+                ))
+            if line.document_type == 'header_footer' and line.form_field.startswith('sol_id_'):
+                raise ValidationError(_(
+                    "Invalid form field name %(field_name)s. A form field name in a header or a"
+                    " footer can not start with \"sol_id_\".",
+                    field_name=line.form_field,
+                ))
+
+    @api.constrains('path')
+    def _check_valid_and_existing_paths(self):
+        """ Verify that the paths exist and are valid.
+
+        :return: None
+        :raises: ValidationError if at least one of the paths isn't valid.
+        """
+        name_pattern = re.compile(r'^(\w|-|\.)+$')
+        for line in self.filtered(lambda line: line.path):
+            if not re.match(name_pattern, line.path):
+                raise ValidationError(_(
+                    "Invalid path %(path)s. It should only contain alphanumerics, hyphens,"
+                    " underscores or points.",
+                    path=line.path,
+                ))
+
+            path = line.path.split('.')
+            is_header_footer = line.document_type == 'header_footer'
+            Model = self.env['sale.order'] if is_header_footer else self.env['sale.order.line']
+            for elem in path:
+                if not Model._fields.get(elem):
```

Comment:
```suggestion
                if elem not in Model._fields:
```
faster, cleaner

Reply:
```suggestion
                if not (field := Model._fields.get(elem)):
```
if you use the field below